### DISCUSSION TITLE
V1.19.1 Works on my Pico W but not on my standard Non Wifi Pico

### DISCUSSION THREAD

**Discussion by pollardd** [micropython/micropython#10738] [RP2040 / Pico]
My project uses two PICO devices that communicate with each other via a LoRa Module.
Testing has been going well in fact I'm almost finished.
[My Project](https://github.com/pollardd/Lora-Gate-Alarm)
Although I don't think my code has anything to do with the problem.

Previously I had the following version of the interpreter running on both devices with the correct WIFI and non WIFI sub versions on each.
MicroPython v1.19.1-850-gfe2a8332f on 2023-02-01; Raspberry Pi Pico W with RP2040

Then V1.19.1 stopped connecting to Thonny when using the non WIFI Pico. 
What did you change? (I hear you ask).
Nothing really, (I hate to say)

If I change back to V1.18 on the non wifi device I can connect no problem from Thonny.
MicroPython v1.18 on 2022-01-17; Raspberry Pi Pico with RP2040
So the USB port and cable are fine.

If I change to a different PC I get the same results.  I can't connect to my non wifi pico v1.19.
The wifi pico works fine on v1.19.

First PC is old Windows7 laptop PC I use just for programming with Thonny.
Second PC I used for testing is a Windows 10 Desktop.

This is the error message I receive.  I have nothing else connected via serial and there are no other versions of Thonny running and if I change back to version 1.18 everything works.
I can't work with version 1.18 for now as I want to use the encryption module "ucryptolib" which doesn't appear to be part of the build until version 1.19.

Any Ideas what's going on here as I'm kind of stuck?

Error Message
```
Unable to connect to COM8: could not open port 'COM8': PermissionError(13, 'Access is denied.', None, 5)
If you have serial connection to the device from another program, then disconnect it there first.
Process ended with exit code 1.
```
**UPDATE**
After struggling with this for more than a day I decided (just for fun) to delete main.py from the Pico using Thonny and V1.18.  Sure enough I could then reinstall V1.19.1 and connect via the serial connection.  I can run main.py from within Thonny.  So far so good.  However when I copy main.py to the PICO again I'm back to where I started.  
So from this I assume something within my code is stopping V1.19.1 from using the "STOP/RESTART BACKEND" button in Thonny to kill main.py.  But why only in V1.19.1?

Main.py is renamed from MainGate.py in my project as it's the one that runs on the non wifi PICO.

**Comment by robert-hh** [comment]
How many files are in the files system of the non-Wifi Pico? There was a problem that at boards with many files registering of the USB port failed. That is fixed mid January. So a  ecent nightly build should not have that problem.

**Reply by pollardd** [reply]
There are 15 files on the non wifi pico including main.py
I did try the latest version of the interpreter that Thonny suggests which is only a day old or something like that.  I had the same result.
How many files is many?  Am I at or near that number?
Thanks
David

**Reply by robert-hh** [reply]
I had the problem with >80 files. So 15 should not be a problem. A one day old version should be recent enough. May it be that the COM port changed, or some other program is holding the COM port? Which com ports do you see in the Windows Device Manager?

**Reply by pollardd** [reply]
It’s always on the same com port. Com8
The fact that I can repeat this on two machines with only the non wifi pico connected tells me it’s not a port conflict.
It’s something to do with the pico running main.py at startup that’s killing Thonny’s ability to connect to the com port.  V1.18 doesn’t have the same problem.  This is looking like a bug in 1.19 to me

**Comment by pollardd** [comment]
I'm not really sure if I should consider this to be an answer.  I guess time will tell.
I downloaded all the interpreter versions available [here ](https://micropython.org/download/rp2-pico/)
Then tried each of them one at a time by dropping the UF2 file straight on the windows explorer drive letter without changing anything else.  The results were mixed.

This version works.  (Last Stable?)
MicroPython v1.19.1 on 2022-06-18; Raspberry Pi Pico with RP2040

This version does not work
rp2-pico-20230203-unstable-v1.19.1-852-g9ea64a36a.uf2

This version does not work
rp2-pico-20230207-unstable-v1.19.1-854-g35524a6fd.uf2

This version works
rp2-pico-20230209-unstable-v1.19.1-855-g9848b0685.uf2

This version works
rp2-pico-20230209-unstable-v1.19.1-859-g41ed01f13.uf2

So it looks like the problem may have been solved in the last two releases both dated 09Feb2023.
Lets see how it goes while I continue debugging the last few features.
Thanks
David
